Implement comments from #428 (#508)
hsf-training/cpluspluscourse: talk/objectorientation/objectsclasses.tex

      // let's square a
      myObj.squareA();
    \end{cppcode*}
      \onslide<2>
    \begin{cppcode*}{gobble=2}
      class MyFirstClass {
        int a;
        public:
        void squareA() {
          a *= a;
        }
        int sum(int b) {
          return a + b;
        }
      };

      MyFirstClass myObj;
      myObj.a = 2; // error

      // let's square a
      myObj.squareA();
    \end{cppcode*}
    \end{overprint}

>       \columnbreak

Reviewer: This will come later, when it's useful. I do not see the added value to repeat it here. It will only take more time to present and blur the message

Author: I will remove this now.

    \center
    \begin{tikzpicture}
      \classbox{MyFirstClass}{
        int a; \\
        void squareA(); \\
        int sum(int b);
      }
    \end{tikzpicture}
    \pause
    \null \vfill
    {
      \setlength\textwidth\columnwidth
    \begin{block}{struct vs.\ class}
      \begin{itemize}

>         \item In structs, every member is accessible
>         \item In classes, members are ``private'' unless made public 

Reviewer 2: ```suggestion
        \item In structs, every member and base is ``public''
        \item In classes, they are ``private'' unless made public 
```

Author: We are just before the slide where we deal with inheritance here. I put this box to not confuse people why we say class al they time, but the keyword is `struct`.

Reviewer 2: Alright, makes sense!

      \end{itemize}
    \end{block}

>     }

Reviewer: Same

\begin{frame}[fragile]
  \frametitlecpp[98]{Implementing methods}
  \begin{goodpractice}{Header/source separation}
    \begin{itemize}
    \item Functions usually implemented in .cpp, outside of class, using the class name as ``namespace''
    \item First include header being implemented, then everything that's needed for implementation
    \item Keep includes and code in headers to a minimum (compiled many times)
      \begin{itemize}
        \item Short member functions can be in the header

>         \item Some functions (templates, \cppinline{constexpr}) must be in the header

Reviewer: Previous version was less verbose and thus more readable. We should try to avoid plain sentences going on several lines and stick to one line bullets. I would also drop this change
